eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTests.java

```diff
@@ -0,0 +1,363 @@
+import com.netflix.discovery.shared.Application;
+import com.netflix.discovery.shared.Applications;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class EurekaEntityFunctionsTests {
+
+    @Test
+    public void testSelectApplicationNamesIfNotNullReturnLastNameAdded() {
+        Applications applications = new Applications();
+        applications.addApplication(new Application("foo"));
+        applications.addApplication(new Application("bar"));
+        applications.addApplication(new Application("baz"));
+
+        HashSet<String> strings = new HashSet<>();
+        strings.add("baz");
+        strings.add("bar");
+        strings.add("foo");
+        Assert.assertEquals(strings,
+                EurekaEntityFunctions.selectApplicationNames(applications));
+    }
+
+    @Test
+    public void testSelectInstancesMappedByIdIFNotNullReturnMapOfInstances() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+
+        Application application = new Application();
+        application.addInstance(instanceInfo);
+
+        HashMap<String, InstanceInfo> hashMap = new HashMap<>();
+        hashMap.put("foo", instanceInfo);
+        Assert.assertEquals(hashMap,
+                EurekaEntityFunctions.selectInstancesMappedById(application));
+    }
+
+    @Test
+    public void testSelectInstanceIfInstanceExistsReturnSelectedInstance() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+        applications.addApplication(application);
+
+        Assert.assertNull(EurekaEntityFunctions.selectInstance(
+                new Applications(), "foo"));
+        Assert.assertNull(EurekaEntityFunctions.selectInstance(
+                new Applications(), "foo", "foo"));
+
+        Assert.assertEquals(instanceInfo, EurekaEntityFunctions.selectInstance(
+                applications, "foo"));
+        Assert.assertEquals(instanceInfo, EurekaEntityFunctions.selectInstance(
+                applications, "foo", "foo"));
+    }
+
+    @Test
+    public void testTakeFirstIfNotNullReturnFirstInstance() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
```

Comment:
Same, this is repetitive and distracts from the test logic, probably best to extract into a method.

Reply:
private method create